### PR TITLE
Modifications for better Swagger doc

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Using seperate models for POST requests
- Added AutoMapper
- Added classes for providing examples for schemas and requestbodies 
- Added authorization for swagger

## Related Issue(s)
- Altinn/altinn-platform#103

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

### CODE REVIEW

File: src/Events/Mappers/SubscriptionMapper.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using Altinn.Platform.Events.Models;
+
+namespace Altinn.Platform.Events.Mappers
+{
+    /// <summary>
+    /// A cøass that holds the subscription mapper configurations
```
Comment (acn-sbuad) [inline src/Events/Mappers/SubscriptionMapper.cs:6]:
```suggestion
    /// A class that holds the subscription mapper configurations
```

**Review by tba76 — APPROVED** [review]
🍺

File: src/Events/Controllers/SubscriptionController.cs [hunk 2]
```diff
@@ -140,7 +146,7 @@ public async Task<ActionResult<string>> Validate(int id)
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
-        public async Task<ActionResult<string>> Delete(int id)
+        public async Task<ActionResult> Delete(int id)
```
Comment (acn-sbuad) [inline src/Events/Controllers/SubscriptionController.cs:149]:
```suggestion
        [Produces("application/json")]
        public async Task<ActionResult> Delete(int id)
```

File: src/Events/Controllers/EventsController.cs [hunk 3]
```diff
@@ -149,8 +170,19 @@ public async Task<ActionResult<string>> Post([FromBody] CloudEvent cloudEvent)
         }
 
         /// <summary>
-        /// Retrieves a set of events based on query parameters.
+        /// Retrieves a set of events related to a party based on query parameters.
         /// </summary>
+        /// <param name="after" example="3fa85f64-5717-4562-b3fc-2c963f66afa6">Id of the latter even that should be included</param>
```
Comment (acn-sbuad) [inline src/Events/Controllers/EventsController.cs:175]:
```suggestion
        /// <param name="after" example="3fa85f64-5717-4562-b3fc-2c963f66afa6">Id of the latter event that should be included</param>
```

File: src/Events/Swagger/RequestBodyExampleFilter.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            {
+                ("Instance created event with alternative subject",
+                CreateOpenApiObject(new List<(string Name, string Value)>()
+                {
+                    ("source", "https://ttd.apps.altinn.no/ttd/apps-test/instances/50015641/a72223a3-926b-4095-a2a6-bacc10815f2d"),
+                    ("specversion",  "1.0"),
+                    ("type",  "app.instance.created"),
+                    ("subject",  "/party/50015677")
+                })),
+                ("Instance crated event without alternative subject",
```
Comment (acn-sbuad) [inline src/Events/Swagger/RequestBodyExampleFilter.cs:49]:
```suggestion
                ("Instance created event without alternative subject",
```

**Pull request merged**